card_generator/cards/client.py

```diff
@@ -0,0 +1,329 @@
+            settings, "PDS_SPP_TRANSLATOR_DEFAULT_FETCH_LIMIT", 100
+        )
+        if not result_params:
+            result_params = {
+                "limit": default_fetch_limit,
+                "offset": 0,
+            }
+        elif "limit" not in result_params:
+            result_params["limit"] = default_fetch_limit
+            result_params["offset"] = 0
+
+        fetched_data = []
+        batch_repetition = (fetch_count - 1) // default_fetch_limit + 1
+        for _ in range(0, batch_repetition):
+            result = self._fetch(
+                model_name, query_params, result_params, related_data=related_data
+            )
+            fetched_data.extend(result)
+            result_params["offset"] += default_fetch_limit
+        return fetched_data
+
+    def _update(
+        self,
+        model_name: str,
+        item_ids: list[int],
+        data: dict,
+        result_params: dict | None = None,
+        *args,
+        **kwargs,
+    ):
+        cleaned_data = self._sanitize_data(
+            model_name=model_name, data=[data], to_pds_spp=True
+        )
+        try:
+            result = self._run_query(
+                model_name=model_name,
+                method_name="write",
+                query_params=[item_ids, cleaned_data[0]],
+                result_params=result_params,
+            )
+        except xmlrpc.client.Fault as e:
+            server_message = e.faultString
+            raise PDSSPPAPIException(
+                f"Something went wrong. Remote server raise an error with logs. Log: {server_message}"
+            )
+        if not result:
+            raise PDSSPPAPIException("Updating data failed.")
+        return self._read(model_name, item_ids, result_params)
+
+    def _run_query(
+        self,
+        model_name: str,
+        method_name: str,
+        query_params: Any | None = None,
+        result_params: Any | None = None,
+    ):
+        server = self.get_server_proxy(f"{self.server_root}{self.MODEL_ENDPOINT}")
+        if not result_params:
+            result_params = {}
+        if query_params is None or query_params == []:
+            query_params = [[]]
+        return server.execute_kw(
+            self.db_name,
+            self.uid,
+            self.password,
+            model_name,
+            method_name,
+            query_params,
+            result_params,
+        )
+
+
+class QueueCardsClient(PDSSPPClient):
+    def get_id_queue_pdfs(self, batch_id: int):
+        id_queue_ids = self.call_api(
+            "fetch",
+            model_name=settings.PDS_SPP_QUEUE_BATCH_MODEL,
+            query_params=[[["id", "=", batch_id]]],
+            result_params={"fields": ["queued_ids"]},
+        )
+
+        return self.call_api(
+            "fetch",
+            model_name=settings.PDS_SPP_ID_QUEUE_MODEL,
+            query_params=[[["id", "in", id_queue_ids]]],
+            result_params={"fields": ["id_pdf"]},
+        )
```

Comment:
Would that not be a too big payload if we do 50 cards?

Reply:
We have some kind of [pagination done here](https://github.com/idpass/idpass-card-generator/pull/44/files#diff-9ad75fdbb6636f05b091973bec36a66f8e881cbcbcecd696bae1dd7b3bebac1dR223) and we can limit the items per fetch [using this config ](https://github.com/idpass/idpass-card-generator/pull/44/files#diff-f7a16228131ae92457c22657bb33458207a7578b8cdb6ff796153916fa67a22dR318)

So when we call `call_api`, it actually fetch the items per batch (default 10) and returns all of them.